### PR TITLE
Make sure NetworkAttachmentDefinition is removed upon purge

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

File: operator/pkg/helmreconciler/prune.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ var (
 	// AllClusterResources lists all cluster scope resources types which should be deleted in purge case, including CRD.
```
Comment (howardjohn) [inline operator/pkg/helmreconciler/prune.go:68]: This is "cluster scoped" but the the yaml we set `namespace`. Seems one is incorrect, not sure which

Comment (jwendell) [reply operator/pkg/helmreconciler/prune.go:68]: It is indeed a namespaced one, not sure then where to put it. I think it should only be removed when purging, as it's mostly static and might be reused across different revisions...

Comment (howardjohn) [reply operator/pkg/helmreconciler/prune.go:68]: hmm now that I looked more, I don't think we ever make any usage of it that is only "Cluster scoped". So may be enough to just rename it..?
Should double check my assumption though

Comment (jwendell) [reply operator/pkg/helmreconciler/prune.go:68]: do you mean rename the `AllClusterResources` variable? To something like `AllResources`?

File: operator/pkg/helmreconciler/prune.go [hunk 2]
```diff
@@ -68,6 +68,7 @@ var (
...
 	AllClusterResources = append(ClusterResources,
 		schema.GroupVersionKind{Group: "apiextensions.k8s.io", Version: "v1", Kind: name.CRDStr},
+		schema.GroupVersionKind{Group: "k8s.cni.cncf.io", Version: "v1", Kind: name.NetworkAttachmentDefinitionStr},
```
Comment (hanxiaop) [inline operator/pkg/helmreconciler/prune.go:71]: Should this be put into the `NamespacedResources`?

Comment (jwendell) [reply operator/pkg/helmreconciler/prune.go:71]: @hanxiaop See my message above to John

**Review by howardjohn — APPROVED** [review]

File: operator/pkg/helmreconciler/prune.go (hunk 2 above)
Comment (hanxiaop) [reply operator/pkg/helmreconciler/prune.go:71]: I see, thanks

**Pull request merged**